server/src/main/java/org/elasticsearch/common/document/DocumentField.java

```diff
@@ -104,7 +118,18 @@ public void writeTo(StreamOutput out) throws IOException {
         if (out.getVersion().onOrAfter(Version.V_7_16_0)) {
             out.writeCollection(ignoredValues, StreamOutput::writeGenericValue);
         }
+        if (out.getVersion().onOrAfter(Version.V_8_2_0)) {
+            out.writeList(lookupFields);
+        } else {
+            if (lookupFields.isEmpty() == false) {
+                assert false : "Lookup fields require all nodes be on 8.2 or later";
```

Comment:
Any reason we both assert and throw here?

Reply:
The assertion is to make sure we never call this method in our tests, and tests always abort if we fail to do so. Without the assertion, Elasticsearch might retry and swallow the error.